calico_kubernetes/calico_kubernetes.py

```diff
@@ -497,217 +540,53 @@ def _api_root_secure(self):
                          'a http or https address. Exiting', KUBE_API_ROOT)
             sys.exit(1)
 
-    def _generate_rules(self, pod):
+    def _generate_profile_name(self):
```

Comment:
I think we can get rid of this.  The only reason this is still here is for the special case handling of the policy-agent pod, which we've determined is the wrong thing to do.

self.profile_name can be set at __init__ time.